src/main/java/cn/nukkit/block/BlockTrapdooDarkOak.java

```diff
@@ -0,0 +1,16 @@
+package cn.nukkit.block;
+
+import cn.nukkit.utils.BlockColor;
+import cn.nukkit.utils.Identifier;
+
+public class BlockTrapdooDarkOak extends BlockTrapdoor {
```

Comment:
These don't need to be separate classes. Instead we could have a factory which passes the block colour and that's it.

Reply:
Interesting, I hadn't thought about that.
Would that be accepted?
```java
this.factoryMap.put(ACACIA_TRAPDOOR, id-> new BlockTrapdoor(id, BlockColor.ORANGE_BLOCK_COLOR)); //400
this.factoryMap.put(BIRCH_TRAPDOOR, id -> new BlockTrapdoor(id, BlockColor.SAND_BLOCK_COLOR)); //401
```
With a new constructor which takes `BlockColor`?

Reply:
I've changed to do the same way as https://github.com/NukkitX/Nukkit/blob/2.0/src/main/java/cn/nukkit/item/PlaceableItem.java